src/server/lib/encryptedStateCookie.ts

```diff
@@ -35,9 +35,17 @@ export const setEncryptedStateCookie = (
 export const readEncryptedStateCookie = (
   req: Request,
 ): EncryptedState | undefined => {
-  const encryptedCookie = baseUri.includes('localhost')
-    ? req.cookies[encryptedStateCookieName]
-    : req.signedCookies[encryptedStateCookieName];
+  // TODO: We need to double check the security implications of using an env variable
+  // to read from the unsigned cookie storage.
+  const encryptedCookie =
+    baseUri.includes('localhost') || process.env.RUNNING_IN_CYPRESS === 'true'
+      ? // If we're in testing or localhost, first try reading from signedCookies,
+        // and only then fall back to regular cookies.
+        Object.keys(req.signedCookies).includes(encryptedStateCookieName)
+        ? req.signedCookies[encryptedStateCookieName]
+        : req.cookies[encryptedStateCookieName]
+      : // If we're not in testing, always read from signedCookies.
+        req.signedCookies[encryptedStateCookieName];
```

Comment:
I find this ternary operator to be super confusing for anything other the most trivial cases. In particular here where there's 2 nested. Wouldn't it be better to just use regular if/else statements or is there a reason to do it this way ?

Reply:
Happy for this to be broken into something more readable!